src/test/java/com/kentyou/featurelauncher/cli/FeatureLauncherCliTest.java

```diff
@@ -0,0 +1,501 @@
+
+		int exitCode = new CommandLine(new FeatureLauncherCli()).execute((args.toString().split(" ")));
+		assertEquals(0, exitCode);
+
+		assertTrue(out.toString().contains(String.format("Launching feature %s", FEATURE_FILE_ID)));
+		assertTrue(out.toString().contains("Using decorators:"));
+		assertTrue(out.toString().contains(String.format("class %s", DECORATOR_CLASS_NAME)));
+
+		assertEquals("", err.toString());
+	}
+
+	@Test
+	public void testDecoratorOptionLong() {
+		StringBuilder args = new StringBuilder();
+		args.append("--dry-run");
+		args.append(" ");
+		args.append("--decorator=");
+		args.append(DECORATOR_CLASS_NAME);
+		args.append(" ");
+		args.append(FEATURE_FILE_PATH);
+
+		int exitCode = new CommandLine(new FeatureLauncherCli()).execute((args.toString().split(" ")));
+		assertEquals(0, exitCode);
+
+		assertTrue(out.toString().contains(String.format("Launching feature %s", FEATURE_FILE_ID)));
+		assertTrue(out.toString().contains("Using decorators:"));
+		assertTrue(out.toString().contains(String.format("class %s", DECORATOR_CLASS_NAME)));
+
+		assertEquals("", err.toString());
+	}
+
+	@Test
+	public void testExtensionHandlerOptionShort() {
+		StringBuilder args = new StringBuilder();
+		args.append("--dry-run");
+		args.append(" ");
+		args.append("-e=");
+		args.append(EXTENSION_HANDLER_NAME);
+		args.append("=");
+		args.append(EXTENSION_HANDLER_CLASS_NAME);
+		args.append(" ");
+		args.append(FEATURE_FILE_PATH);
+
+		int exitCode = new CommandLine(new FeatureLauncherCli()).execute((args.toString().split(" ")));
+		assertEquals(0, exitCode);
+
+		assertTrue(out.toString().contains(String.format("Launching feature %s", FEATURE_FILE_ID)));
+		assertTrue(out.toString().contains("Using extension handlers:"));
+		assertTrue(out.toString()
+				.contains(String.format("%s = class %s", EXTENSION_HANDLER_NAME, EXTENSION_HANDLER_CLASS_NAME)));
+
+		assertEquals("", err.toString());
+	}
+
+	@Test
+	public void testExtensionHandlerOptionLong() {
+		StringBuilder args = new StringBuilder();
+		args.append("--dry-run");
+		args.append(" ");
+		args.append("--extension-handler=");
+		args.append(EXTENSION_HANDLER_NAME);
+		args.append("=");
+		args.append(EXTENSION_HANDLER_CLASS_NAME);
+		args.append(" ");
+		args.append(FEATURE_FILE_PATH);
+
+		int exitCode = new CommandLine(new FeatureLauncherCli()).execute((args.toString().split(" ")));
+		assertEquals(0, exitCode);
+
+		assertTrue(out.toString().contains(String.format("Launching feature %s", FEATURE_FILE_ID)));
+		assertTrue(out.toString().contains("Using extension handlers:"));
+		assertTrue(out.toString()
+				.contains(String.format("%s = class %s", EXTENSION_HANDLER_NAME, EXTENSION_HANDLER_CLASS_NAME)));
+
+		assertEquals("", err.toString());
+	}
+
+	@Test
+	public void testLaunchPropertyOptionShort() {
+		StringBuilder args = new StringBuilder();
+		args.append("--dry-run");
+		args.append(" ");
+		args.append("-l=");
+		args.append(LAUNCH_PROPERTY_1_KEY);
```

Comment:
The logic in this and subsequent tests is all correct, but I don't see any testing for 

    -l key1=value1,key2=value2,keyN=valueN

The same comma-separated versions are present for launch properties, variables and configurations in the specification. I have now noticed that they are missing for extension handlers and decorators - what would your opinion be about adding them there?

Reply:
Regarding testing for multiple extension handlers and multiple decorators - this is only possible once more than one decorator / more than one extension handler exists. Currently there is only one of each, and I cannot pass just any class, it must be class which implements `FeatureDecorator` / `FeatureExtensionHandler` respectively.